modules/edge-identity/src/identity.rs

```diff
@@ -225,11 +240,14 @@ impl<T: Trait> Module<T> {
 		let id_type = record.identity_type.encode().clone();
 		let id = record.identity.encode().clone();
 		if approve {
+			// Add the registration bond amount on behalf of a successful verification
+			T::Currency::deposit_into_existing(&record.account, Self::registration_bond()).unwrap();
+			// Remove identity from list of pending identities
 			<IdentitiesPending<T>>::mutate(|idents| idents.retain(|(hash, _)| hash != identity_hash));
 			Self::deposit_event(RawEvent::Verify(*identity_hash, sender, id_type, id));
 			<IdentityOf<T>>::insert(identity_hash, IdentityRecord {
 				stage: IdentityStage::Verified,
-				expiration_time: T::Moment::zero(),
+				expiration_length: T::BlockNumber::zero(),
 				..record
 			});
 		} else {
```

Comment:
In this else case, if you want to slash the reserved amount as punishment, you can use the slash_reserved function https://github.com/paritytech/substrate/blob/master/srml/council/src/seats.rs#L165

Reply:
And what if the reserve has already been slashed from something else?

Reply:
I feel like it's safe to assume that other modules wont randomly slash reserved balance that they didn't themselves reserve...